File: QuSoC/QuSoC/apps/Increment/firmware/Increment.cpp
#include "Increment.h"
#include <stdint.h>
#include <stdbool.h>
#include "soc.h"
namespace Increment
{
	void Firmware::EntryPoint()
	{
		uint32_t counter = 0;
		while(true)
		{
			(counter++);
			Increment_SOC_Counter = counter;
		}
	}
}

File: QuSoC/QuSoC/apps/Increment/source/Increment.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Increment
{
    public class SOC
    {
        public static SOC Instance { get; set; } = new SOC();

        public virtual uint Counter { get; set; }
    }

    public static class Firmware
    {
        public static void EntryPoint()
        {
            uint counter = 0;
            while (true)
            {
                counter++;
                SOC.Instance.Counter = counter;
            }
        }
    }
}
